File: swissmedhealth/swissmedhealth/web_form/gdpr_consent/gdpr_consent.js
frappe.ready(function () {

	let params = new URLSearchParams(window.location.search);
	let email_id = params.get('email_id');
	if (email_id) {
		frappe.call('swissmedhealth.swissmedhealth.web_form.gdpr_consent.gdpr_consent.get_customer_consent_details', { email_id: email_id }).then(r => {
			let doc = r.message;
			
			// if (doc.acceptance_date) is empty set the current date
			if (!doc.acceptance_date) {
				doc.acceptance_date = frappe.datetime.nowdate();
			}

			// set the values in the form
			frappe.web_form.set_values(doc);
			frappe.web_form.is_new = false;
			frappe.web_form.doc.name = doc.name;			
		});
	} else {
		// redirect to home page
		window.location.href = '/';
	}

	// bind events here
	$('.submit-btn').on('click', function (e) {
		// Prevent the default form submission
		e.preventDefault();

		frappe.call('swissmedhealth.swissmedhealth.web_form.gdpr_consent.gdpr_consent.save', { doc: frappe.web_form.doc }).then(() => {
			frappe.msgprint({
				title: __('Success'),
				indicator: 'green',
				message: __('We have successfully received your medical history. A member of our professional team will be in touch with you shortly to discuss the next steps. Thank you for entrusting us with your healthcare needs.')
			});

			$('.web-form-container').hide();
		}).catch((err) => {
			frappe.msgprint({
				title: __('Error'),
				indicator: 'red',
				message: __('An error occurred while submitting your details. Please try again later.')
			});
		});
	});

})

File: swissmedhealth/swissmedhealth/web_form/gdpr_consent/gdpr_consent.py
import frappe

def get_context(context):
	# do your magic here
	pass

@frappe.whitelist(allow_guest=True)
def get_customer_consent_details(email_id):
	lead = frappe.db.get_value("Lead", {"email_id": email_id}, ["custom_customer_consent"], as_dict=True)
	# if empty lead throw error
	if not lead:
		frappe.throw("Not found")
	
	# get the dental history document from the lead
	customer_consent = frappe.get_value("Customer Consent", lead.custom_customer_consent, ["*"], as_dict=True)
	return customer_consent

@frappe.whitelist(allow_guest=True)
def save(doc):
	# doc is in json format. Convert it to dict
	doc = frappe.parse_json(doc)
	# Get the lead from the doc.name assign all values from doc to lead and save
	lead = frappe.get_doc("Customer Consent", doc.name)
	lead.update(doc)
	lead.save(ignore_permissions=True)
	return lead